rr_evgp/src/UniformCostSearch/UniformCostSearch.cpp

```diff
@@ -0,0 +1,126 @@
+#include <rr_evgp/UniformCostSearch.h>
+#include <algorithm>
+#include <functional>
+#include <opencv2/core/mat.hpp>
+#include <opencv2/core/types.hpp>
+#include <queue>
+#include <vector>
+
+UniformCostSearch::UniformCostSearch(cv::Mat obstacleGrid, cv::Mat distanceGrid, cv::Point startPt, cv::Point goalPt) {
+    obstacleGrid_ = obstacleGrid;
+    distanceGrid_ = distanceGrid;
+    startPoint_ = startPt;
+    goalPoint_ = goalPt;
+}
+
+UniformCostSearch::State UniformCostSearch::getStartState() {
+    UniformCostSearch::State s;
+    s.pt = startPoint_;
+    s.cost = 0.0;
+    return s;
+}
+
+bool UniformCostSearch::isGoalState(UniformCostSearch::State s) {
+    if (s.pt == goalPoint_) {
+        return true;
+    }
+    return false;
+}
+
+float UniformCostSearch::pointToCost(cv::Point pt) {
+    return distanceGrid_.at<float>(pt);  // DistanceTransform outputs CV_32F
+}
+
+bool UniformCostSearch::isValidPoint(cv::Point pt) {
+    cv::Rect rect(cv::Point(), distanceGrid_.size());
```

Comment:
Don't think it would make much of a difference but this could be a global variable or static so you didn't have to create it for every call